zookeeper-assembly/src/main/assembly/bin-package.xml

```diff
@@ -29,29 +29,23 @@
     <componentDescriptor>src/main/assembly/components.xml</componentDescriptor>
   </componentDescriptors>
 
-  <moduleSets>
-    <!-- ZooKeeper jars (excluding pom projects) including 3rd party dependencies -->
-    <moduleSet>
-      <useAllReactorProjects>true</useAllReactorProjects>
+  <dependencySets>
+    <dependencySet>
       <includes>
-        <include>org.apache.zookeeper:zookeeper</include>
+        <include>*:*</include>
       </includes>
-
-      <binaries>
-        <outputDirectory>lib</outputDirectory>
-        <unpack>false</unpack>
-        <dependencySets>
-          <dependencySet>
-            <excludes>
-              <exclude>org.apache.zookeeper:zookeeper-recipes</exclude>
-              <exclude>org.apache.zookeeper:zookeeper-client</exclude>
-              <exclude>org.apache.zookeeper:zookeeper-docs</exclude>
-            </excludes>
-          </dependencySet>
-        </dependencySets>
-      </binaries>
-    </moduleSet>
-  </moduleSets>
+      <excludes>
+         <exclude>org.apache.zookeeper:zookeeper-recipes</exclude>
```

Comment:
With the changes in the bin-package I get a warning when building teh tarball:
"Cannot include project artifact: org.apache.zookeeper:zookeeper-assembly:pom:3.6.0-SNAPSHOT; it doesn't have an associated file or directory."

What was the reason to change moduleSets to dependencySets?

Also a super-nit: excludes have 3 space indents here.

Reply:
I can fix that warning.
I had to change from modueleSet to dependencySets in order to have morw control (in my experience) over the list of jars to include (and with moduleSet I was not able to include the 'provided' dependencies).

I can't fix the warning now, if Andor wants to commit this change soon you can pick up the branch, I think we can just add exclusion or I can do it tonight